### ISSUE TITLE
Feature: hacktoberfest activity graph should only show data from oct 1st - oct 31st

### ISSUE THREAD

**Issue by jpmcb** [open-sauced/app#2144]
### Type of feature

🍕 Feature

### Current behavior

The activity chart for Hacktoberfest looks strange since the data returned by the API for the `hacktoberfest` topic only return data for the configured hacktoberfest year:

![Screen Shot 2023-11-14 at 2 23 08 PM](https://github.com/open-sauced/app/assets/23109390/aea0da85-889b-4c3f-a451-2295238e24fe)

So this chart is _technically_ correct since the API isn't returning data from a few days ago, only data from Oct 1st - Oct 31st.

cc @bdougie noticed this on stream today

### Suggested solution

The activity chart should only show data from Oct 1st - Oct 31st when the topic is "hacktoberfest", similar to how the API returns the data.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

**Comment by takanome-dev** [comment]
> The activity chart should only show data from Oct 1st - Oct 31st when the topic is "hacktoberfest", similar to how the API returns the data.

I think the data is correct but there is an overflow happening here probably due to the position of the chart data in the x axis

![chart](https://github.com/open-sauced/app/assets/79809121/688f6587-6d36-4efc-ad2e-39e97c553c3a)

Maybe we should fix the overflow and change the default topic here.

**Comment by jpmcb** [comment]
Exactly: the data is correct, it just shouldn't be based on _"now"_, it should be solely based on Oct 1st - Oct 31st when the topic is "hacktoberfest"

**Comment by a0m0rajab** [comment]
Adding the link: https://beta.app.opensauced.pizza/hacktoberfest/activity/filter/top-100-repos?range=90

and screenshot from today; 
![image](https://github.com/open-sauced/app/assets/18273833/6504dae7-2605-4588-bdfc-05835810d2b2)

**Comment by a0m0rajab** [comment]
Related to #2372
